### PR TITLE
feat: split login

### PR DESCRIPTION
**Pull request by rephus**
Closes: #9315

### Description:

With existing google open id  javier@lightdash.com

Screenshot from 2024-03-12 16-43-27

with existing (fake) okta open id 

Screenshot from 2024-03-12 16-43-39

without open id 
Screenshot from 2024-03-12 16-43-47

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by rephus** [comment]
Something weird is happening when I have a sso account and I type my email (even before clicking `continue` ) I don't see an options request

[Screencast from 15-03-24 08:43:00.webm](https://github.com/lightdash/lightdash/assets/1983672/a25bb362-b98e-47a2-b646-5b2c92f9316d)

**Comment by rephus** [comment]
One more thing, I disabled my password locally my screen flashed, but no error showed up, I should get an error. perhaps errors are not handled correctly on the login query.

I'm testing this here https://github.com/lightdash/lightdash/pull/9351 

[Screencast from 15-03-24 09:09:05.webm](https://github.com/lightdash/lightdash/assets/1983672/106c1c1d-7f3f-4250-8867-4d79835dedad)

**Comment by rephus** [comment]
Ok, this is weird, not sure we can do about it . Because of the feature flag, first we get the first form,a nd then we get the other. I know this is a temporary issue, but may be we can just not show anything until the flag is loaded... 

[Screencast from 15-03-24 09:24:09.webm](https://github.com/lightdash/lightdash/assets/1983672/69c376ba-3831-4a61-b23b-4307cc66c819)

**Comment by magnew** [comment]
> Something weird is happening when I have a sso account and I type my email , redirection doesn't work on the first `continue`, then typing my email reloads the page.
> 
>  Screencast.from.15-03-24.08.46.11.webm

I'll see what I missed here

**Comment by magnew** [comment]
> r I think this could be a redir

I'll look at this too.

**Comment by magnew** [comment]
> Ok, this is weird, not sure we can do about it . Because of the feature flag, first we get the first form,a nd then we get the other (sometimes it takes a few seconds !) . I know this is a temporary issue, but may be we can just not show anything until the flag is loaded...
> 
>  Screencast.from.15-03-24.09.24.09.webm

I saw this and figured maybe it wasn't worth the effort to fix since its a temporary flagging issue. But let me see if there is something obvious to do.

File: packages/frontend/src/features/users/components/LoginLanding.tsx [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+        loginOptionsSuccess && loginOptions?.showOptions ? 'login' : 'precheck';
+
+    const isEmailLoginAvailable =
+        loginOptions?.showOptions &&
+        loginOptions?.showOptions.includes(LocalIssuerTypes.EMAIL);
+
+    const disableControls =
+        (loginOptionsLoading && loginOptionsFetched) || isLoading || isSuccess;
+
+    const googleAuthAvailable = health.data?.auth.google.enabled;
```
Comment (magnew) [inline packages/frontend/src/features/users/components/LoginLanding.tsx:129]: Is there a better way to do this?

Comment (rephus) [reply packages/frontend/src/features/users/components/LoginLanding.tsx:129]: no, that's the best way

**Review by magnew — APPROVED** [review]
Tested

**Pull request merged**